pkg/kubefed2/federate/federate.go

```diff
@@ -183,32 +185,68 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 	}, nil
 }
 
-func lookupTypeDetails(hostConfig *rest.Config, typeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
-	apiResource, err := enable.LookupAPIResource(hostConfig, typeName.Name, "")
+func getTypeConfig(hostConfig *rest.Config, typeName, namespace string, outputYAML bool) (typeconfig.Interface, error) {
+	// Lookup kubernetes API availability
+	apiResource, err := enable.LookupAPIResource(hostConfig, typeName, "")
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "Failed to find target API resource %s", typeName)
 	}
-	typeName.Name = typeconfig.GroupQualifiedName(*apiResource)
+	glog.V(2).Infof("API Resource for %s found", typeName)
 
-	client, err := genericclient.New(hostConfig)
+	typeConfig, err := attemptToLookupFederatedType(hostConfig, *apiResource, namespace)
+	if err == nil {
+		return typeConfig, nil
+	}
+	if apierrors.IsNotFound(err) && !outputYAML {
+		return nil, errors.Errorf("%v. Try 'kubefed2 enable type %s' before federating the resource", err, typeName)
+	}
+	if outputYAML {
+		glog.V(1).Infof("Lookup of installed type failed: %v - Falling back on generated default values instead.", err)
```

Comment:
Please avoid embedding an error in the middle of a message, and prefer appending it to the end e.g.

```
Log("Falling back to a generated type config due to lookup failure: %v", err)
```

Reply:
As in previous comment, I did embed the error in between because it did just read better to me.
Updating this anyways as this sequence also looks ok.

Reply:
I take your point about the NotFound error being embeddable, but in cases where there is no specific error (as is the case here), the error message should almost always be placed at the end of a log message.  That ensures that a really long message doesn't result in suffixed content being confused with the error content.